### PR TITLE
Authentication + spatialconnect redux improvements

### PR DESCRIPTION
**Pull request by frankrowe**
## Status
**IN DEVELOPMENT**

## Description
- Adds authentication (TODO: Add Signup form)
- Changes to spatialconnect library allow subscriptions to be dispatched directly to redux store
- general cleaning + refactoring

## Todos
- [x] Tests
- [x] Documentation

```sh
git pull --prune
git checkout auth
react-native run-ios
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: app/SCMobile.js [hunk 1]
```diff
@@ -26,16 +30,51 @@ import reducer from './reducers';
...
+//     sc.xAccessToken$().subscribe(token => {
+//       console.log('token', token);
+//       store.dispatch({
+//         type: 'LOGIN_USER_SUCCESS',
+//         token: token
+//       });
+//     });
+//   }
+// });
+
```
Comment (tetriscode) [inline app/SCMobile.js]: remove this

File: app/SCMobile.js [hunk 2]
```diff
@@ -26,16 +30,51 @@ import reducer from './reducers';
...
 class SCMobile extends Component {
   constructor(props) {
     super(props);
   }
 
   componentWillMount() {
     sc.startAllServices();
+    sc.loginStatus$().subscribe(status => {
```
Comment (tetriscode) [inline app/SCMobile.js]: .do(function (x) { console.log(x); })

File: app/components/SCTest.js [hunk 3]
```diff
@@ -94,13 +99,15 @@ class SCTest extends Component {
...
+      sc.createFeature$(sf.serialize())
+      .map(action => action.payload)
+      .subscribe((data) => {
         try {
           data.should.be.a.Object();
           data.id.should.be.a.String();
           done(true);
-          this.deleteTest(data);
-          this.updateTest(data);
+          //this.deleteTest(data);
```
Comment (marcusthebrown) [inline app/components/SCTest.js]: Should we just delete these lines instead of commenting them out?

File: app/containers/AuthComponent.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React, { Component, PropTypes } from 'react';
+import { View } from 'react-native';
+import { connect } from 'react-redux';
+import { Actions } from 'react-native-router-flux';
+import LoginView from '../components/LoginView';
+
+export const requireAuth = (Component) => {
```
Comment (marcusthebrown) [inline app/containers/AuthComponent.js]: I like this pattern of a wrapped auth component.  Gold star for you ⭐

File: app/reducers/auth.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+import * as sc from 'spatialconnect/native';
+
+const initialState = {
+  token: null,
+  userName: null,
```
Comment (marcusthebrown) [inline app/reducers/auth.js]: Where would this get set?  I don't see it in the reducers so maybe we should remove it for now?

**Pull request merged**